harness/determined/pytorch/experimental/_torch_batch_process.py

```diff
@@ -24,12 +24,21 @@
 class TorchBatchProcessorContext(pytorch._PyTorchReducerContext):
     def __init__(self, core_context: core.Context, storage_path: str) -> None:
         super().__init__()
-        self._core_context = core_context
-        self.distributed = self._core_context.distributed
+        self.core_context = core_context
+        self.distributed = self.core_context.distributed
```

Comment:
why is this being made public?

Reply:
Because the user needs it to report metrics and I would prefer not to ask them to use an underscore function or have to implement some specific metrics reporting function for `TorchBatchProcessor`

Reply:
Exposing an entire low-level API to users of a higher-level API is not the right way to go though.

The Core API offers a ton of functionality, some of it more or less required, and when you hand the end user a core api context then suddenly it's not clear at all who's responsible for which parts of the core api.  Who handles preemption?  Does the user, now that they have core_context.preemption?  Who talks to the searcher api with core_context.searcher?

If you want to expose something to consumers of torch_batch_process, you should be exposing specifically what you want to give to the user in `TorchBatchProcessorContext`, not giving them access to a whole lower layer.

This is how we've done this with our existing high-level APIs ([example](https://github.com/determined-ai/determined/blob/5c839013e43f06365a4a5a0af1a37c8d99f588de/harness/determined/pytorch/_pytorch_context.py#L983-L987)).

Reply:
hid `core_context` and `distributed`. The relevant functions are now exposed directly through the `TorchBatchProcessorContext`